test/client/multi_percolate_test.rb

```diff
@@ -26,6 +26,18 @@
           }
         }
 
+      # COMPATIBILITY
+      # percolator clause must now be defined in index mappings as ordinary data type for ES versions >= 5.x
+      if es_version_5_x?
+          @index.update_mapping("percolator",
+            :percolator => {
+              :properties => {
+                :query => { :type => "percolator" }
+              }
+            }
+          )
+      end
```

Comment:
This might create a race condition between the `create` call above and then the `udpate_mapping` call here. Another approach would be to create a `mappings` hash and then add the percolator section when testing under ES 5.X.

```ruby
mappings = {
   # doc1 and doc2 go here
}

if es_version_5_x?
  mappings[:percolator] = { ... }
end

@index.create \
  :settings => { ... },
  :mappings => mappings
```

Or something like that.

Reply:
excellent, will do!